### PR TITLE
[tlse] tls for KeystoneAPI pod configuration

### PR DESCRIPTION
**Pull request by stuggi**
Public/Internal service cert secrets and the CA bundle secret can be passed to configure httpd virtual hosts for tls termination. The CA cert get direct mounted as the environment bundle to /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem . The service certificates like config files and copied via kolla to /etc/pki/tls/certs/%s.crt|/etc/pki/tls/private/%s.key .
Job deployments for bootstrap/cron get the CA bundle added if configured.

Also indexes the named input resources for password, CA bundle, and endpoint secrets to be able to watch them for a change and reconcile.

Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/428

Jira: [OSPRH-2183](https://issues.redhat.com//browse/OSPRH-2183)

### CODE REVIEW

File: controllers/keystoneapi_controller.go [hunk 1]
```diff
@@ -1015,6 +1189,21 @@ func (r *KeystoneAPIReconciler) generateServiceConfigMaps(
 		"TransportURL":     string(transportURLSecret.Data["transport_url"]),
 	}
 
+	// create httpd  vhost template parameters
+	httpdVhostConfig := map[string]interface{}{}
+	for _, endpt := range []service.Endpoint{service.EndpointInternal, service.EndpointPublic} {
+		endptConfig := map[string]interface{}{}
+		endptConfig["ServerName"] = fmt.Sprintf("keystone-%s.%s.svc", endpt.String(), instance.Namespace)
```
Comment (olliewalsh) [inline controllers/keystoneapi_controller.go:1196]: keystone.ServiceName might be better than hard-coding 'keystone'

File: templates/keystoneapi/config/httpd.conf [hunk 2]
```diff
@@ -22,24 +22,40 @@ SetEnvIf X-Forwarded-For "^.*\..*\..*\..*" forwarded
...
+  SSLEngine on
+  SSLCertificateFile      "{{ $vhost.SSLCertificateFile }}"
+  SSLCertificateKeyFile   "{{ $vhost.SSLCertificateKeyFile }}"
+{{- end }}
 
   ## WSGI configuration
   WSGIApplicationGroup %{GLOBAL}
-  WSGIDaemonProcess keystone display-name=keystone group=keystone processes=3 threads=1 user=keystone
-  WSGIProcessGroup keystone
+  WSGIDaemonProcess {{ $endpt }} display-name={{ $endpt }} group=keystone processes=6 threads=1 user=keystone
```
Comment (olliewalsh) [inline templates/keystoneapi/config/httpd.conf:56]: why are number processes changing?

File: controllers/keystoneapi_controller.go (hunk 1 above)
Comment (stuggi) [reply controllers/keystoneapi_controller.go]: done

**Review by olliewalsh — APPROVED** [review]
/lgtm

**Pull request merged**